scipy/stats/_stats_py.py

```diff
@@ -4363,6 +4371,10 @@ def confidence_interval(self, confidence_level=0.95, method=None):
         return ci
 
 
+xp_capabilities_table['pearsonr'] = dict(cpu_only=True, exceptions=['cupy'])
```

Comment:
I'm not sure I have a strong opinion but I wonder if we should should use the function handle as the key instead as, for example, `linalg` and `sparse.linalg` have functions with the same name

Reply:
The other example shows why we don't necessarily just want one key per function, and in that case strings are appropriate. Collisions will be caught during testing, and the user is welcome to fix them by providing a fully qualifiied name as a string or function handle, if they prefer.

Reply:
Fair point. Another advantage of using the function handle is that it would allow for a module level comparability table in the docs but I guess we can always change things down the line if that's something we want to add